src/server/connection.rs

```diff
@@ -73,5 +73,10 @@ impl Connection {
                 }
             }
         }
+
+        let _ = left_tx.close().await;
+        let _ = right_tx.close().await;
+
```

Comment:
just for good measure, probably not necessary since it goes out of scope anyway.